### ISSUE TITLE
Windows tray installation fails

### ISSUE THREAD

**Issue by adrianriobo** [code-ready/crc#1701]
### General information

  * OS: Windows 10 Pro
  * Hypervisor:Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version
```bash
CodeReady Containers version: 1.19.0+94b0362
OpenShift version: 4.6.3 (embedded in executable)
```
  
## CRC status
```bash
Machine 'crc' does not exist. Use 'crc start' to create it
```

## CRC config
```bash
- pull-secret-file                      : XXXX
```

## Host Operating System
```bash
Host Name:                 CRCQE-WIN10
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.18363 N/A Build 18363
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
...
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. crc setup --enable-experimental-features
  2. enter the user password 

### Expected

Windows tray should be installed

### Actual

Got a fail trying to install the tray, some problem with the temporary script for installation 

### Logs

INFO Checking if podman remote executable is cached
INFO Checking if goodhosts executable is cached
INFO Checking if CRC bundle is cached in '$HOME/.crc'
INFO Checking minimum RAM requirements
INFO Checking if running as normal user
INFO Checking Windows 10 release
INFO Checking Windows edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
INFO Checking if tray executable is present
INFO Checking if tray is installed
INFO Installing CodeReady Containers tray
? Enter account login password for service installation [? for help] *******                                            INFO Will run as admin: Installing System Tray for CodeReady Containers
Installation script didn't execute successfully: CreateFile C:\Users\crcqe\AppData\Local\Temp\crc121645543\success: The system cannot find the file specified.

**Comment by gbraad** [comment]
can you check if `C:\Users\crcqe\AppData\Local\Temp\crc121645543` got created?

**Comment by adrianriobo** [comment]
No, the folder does not exist.

**Comment by gbraad** [comment]
This is supposed to be a temporary folder that gets created in a user writable area. it is best to verify this immediately afte the run as it could have been cleared already.

We nneed this to store a placeholder/marker to indicate the elevated privileges were succesful or not. So eiither the perrmissions were denied for this temp folder creation, or the steps beforehand really failed. In either case, best to include a debug log

**Comment by adrianriobo** [comment]
Thank you for the clarification; 

I checked the folder right after the execution, not when you asked me but when I got the error (as it was the first think I thought to check), anyway I executed it back again in case I missed something the first time (so double checked: fail and folder does not exist right after the execution.

On the other hand I also checked logs with debug, that is true it adds a couple of traces but nothing related with the temporary folder, so as there was no difference there I assumed they were not relevant. Anyway here is the log with the debug level:

```powershell
PS C:\Users\crcqe> crc setup --enable-experimental-features --log-level debug                                           DEBU CodeReady Containers version: 1.19.0+94b0362
DEBU OpenShift version: 4.6.3 (embedded in executable)
INFO Checking if podman remote executable is cached
DEBU Currently podman remote is not supported
INFO Checking if goodhosts executable is cached
DEBU goodhost executable already cached
INFO Checking if CRC bundle is cached in '$HOME/.crc'
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34186895360 bytes
INFO Checking if running as normal user
INFO Checking Windows 10 release
INFO Checking Windows edition
DEBU Running on Windows Professional edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
INFO Checking if tray executable is present
INFO Checking if tray is installed
DEBU Ignoring check and forcing installation of System Tray
INFO Installing CodeReady Containers tray
? Enter account login password for service installation [? for help] *******                                            INFO Will run as admin: Installing System Tray for CodeReady Containers
Installation script didn't execute successfully: CreateFile C:\Users\crcqe\AppData\Local\Temp\crc216549775\success: The system cannot find the file specified.
```

**Comment by anjannath** [comment]
The temporary directory is removed after the setup happens, so it'll not be possible to check for the actual script without modifying the code to not remove the dir after this setup preflight check has run.

https://github.com/code-ready/crc/blob/94b0362a6c6ed753c634b4d7970a3548bb9d9eb9/pkg/crc/preflight/preflight_checks_tray_windows.go#L45-L48

I think commit 71088d7771f0232ebfef06f496ec2f325e769eac will fix this, have tested it, but need to verify it again!

**Comment by thinko** [comment]
This is still a current issue in 1.21.0 and 1.20.0...

With the 1.20.0 installer:
```K:\crc\crc-windows-1.20.0-amd64>crc setup --enable-experimental-features --log-level debug
DEBU CodeReady Containers version: 1.20.0+ef3f80d
DEBU OpenShift version: 4.6.6 (embedded in executable)
INFO Checking if podman remote executable is cached
DEBU Currently podman remote is not supported
INFO Checking if goodhosts executable is cached
DEBU goodhost executable already cached
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34305413120 bytes
INFO Checking if running as normal user
INFO Checking Windows 10 release
INFO Checking Windows edition
DEBU Running on Windows Professional edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
INFO Checking if tray executable is present
INFO Checking if tray is installed
DEBU Ignoring check and forcing installation of System Tray
INFO Installing CodeReady Containers tray
? Enter account login password for service installation [? for help] **************
INFO Will run as admin: Installing System Tray for CodeReady Containers
Installation script didn't execute successfully: CreateFile C:\Users\ahand\AppData\Local\Temp\crc838561755\success: The system cannot find the file specified.
```

and with the 1.21.0 installer:
```C:\Dev\crc121>crc setup --enable-experimental-features --log-level debug
DEBU CodeReady Containers version: 1.21.0+68a4cdd7
DEBU OpenShift version: 4.6.9 (embedded in executable)
DEBU Running 'crc setup'
INFO Checking if podman remote executable is cached
DEBU Currently podman remote is not supported
INFO Checking if admin-helper executable is cached
DEBU Running 'C:\Users\ahand\.crc\bin\admin-helper-windows.exe --version'
DEBU Found admin-helper-windows.exe version 0.0.2
DEBU admin-helper executable already cached
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34305413120 bytes
INFO Checking if running as normal user
INFO Checking Windows 10 release
INFO Checking Windows edition
DEBU Running on Windows Professional edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
INFO Checking if tray executable is present
INFO Checking if tray is installed
DEBU Ignoring check and forcing installation of System Tray
INFO Installing CodeReady Containers tray
? Enter account login password for service installation [? for help] **************
INFO Will run as admin: Installing System Tray for CodeReady Containers
Installation script didn't execute successfully: CreateFile C:\Users\ahand\AppData\Local\Temp\crc900730823\success: The system cannot find the file specified.
```

**Comment by thinko** [comment]
Is there any way to skip the systray service installation?

**Comment by cfergeau** [comment]
Seems like there are no `skip-xxx` options to skip it .

**Comment by craigjperry2** [comment]
I've been running into this same failure, I haven't quite figured it out yet.

```
❯ .\crc.exe setup --enable-experimental-features --log-level=debug
DEBU CodeReady Containers version: 1.21.0+68a4cdd7
DEBU OpenShift version: 4.6.9 (embedded in executable)
DEBU Running 'crc setup'
INFO Checking if podman remote executable is cached
DEBU Currently podman remote is not supported
INFO Checking if admin-helper executable is cached
DEBU Running 'C:\Users\craig\.crc\bin\admin-helper-windows.exe --version'
DEBU Found admin-helper-windows.exe version 0.0.2
DEBU admin-helper executable already cached
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34285166592 bytes
INFO Checking if running as normal user
INFO Checking Windows 10 release
INFO Checking Windows edition
DEBU Running on Windows Professional edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
INFO Checking if tray executable is present
INFO Checking if tray is installed
DEBU Ignoring check and forcing installation of System Tray
INFO Installing CodeReady Containers tray
INFO Will run as admin: Installing System Tray for CodeReady Containers *********
Installation script didn't execute successfully: CreateFile C:\Users\craig\AppData\Local\Temp\crc641201395\success: The system cannot find the file specified.
```

After trying a few different things, multiple reboots and what not, i decided to try on a fresh windows install. I was changing my ssd, i wouldn't just do that for this problem 😉 

Anyway, I did a fresh install of windows 10 pro on the new ssd and once all updates were applied i ran crc setup --enable-experimental-features. I had to reboot obviously because hyper-v wasn't activated yet, reran setup and getting the same failure again.

I'll have a look through the code and see if i can spot what's happening.

All the best,

Craig

Winver: 20H2 OS Build 19042.789

**Comment by craigjperry2** [comment]
As a workaround, i was able to get this running by

```
.\crc config set enable-experimental-features false
.\crc.exe setup --log-level=debug
```

I was able to crc start after that.

**Note** according to my debug logs i'm running v1.21 however the output of my logging matches the strings in the preflight_windows.go at v1.20 tag in this git repo. master and the v1.21 branch would log forcing installation of daemon whereas my crc cli said forcing installation of System Tray - even though it claims to be a v1.21 release. Possible issue in packaging / building of crc for windows?

**Comment by craigjperry2** [comment]
This is now fixed for me in 

```
❯ crc version
CodeReady Containers version: 1.22.0+6faff76f
OpenShift version: 4.6.15 (embedded in executable)
```

**Comment by anjannath** [comment]
We have changed the way the tray is installed, its still under experimental features (i.e `crc setup --enable-experimental-features`)
but the issue reported here should be fixed, does it also work for you now @thinko

**Comment by cfergeau** [comment]
Closing this as tray installation is very different now. Please open a new issue if you are still seeing this.
